### PR TITLE
Merge back master

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Comment by crtrott** [comment]
Is this stupid to do?

**Comment by masterleinad** [comment]
What's the rationale for having all these commits in the commit history?
I would just keep the resulting changes updating the version.

**Comment by crtrott** [comment]
the rational would be to merge the histories again. Essentially we have divergent histories, which I think might make bisecting for example harder.

**Comment by masterleinad** [comment]
Hmmm... For 3.0 we also only had one commit: https://github.com/kokkos/kokkos/pull/2730/files and I still don't really see how having all these commits would help with bisecting a lot.

File: master_history.txt [hunk 1]
```diff
@@ -19,3 +19,4 @@ tag:  2.8.00     date: 02:05:2019    master: 34931a36    develop: d1659d1d
 tag:  2.9.00     date: 06:24:2019    master: 5d6e7fb3    develop: 4c6cb80a
 tag:  3.0.00     date: 01:31:2020    master: 2983b80d    release-candidate-3.0: fdc904a6
 tag:  3.1.00     date: 04:14:2020    master: cd1b1d0a    develop: fd90af43
+tag:  3.1.1      date: 05:04:2020    master: 785d19f2    release: 2be028bc
```
Comment (dalg24) [inline master_history.txt:22]:
```suggestion
tag:  3.1.01     date: 05:04:2020    master: 785d19f2    release: 2be028bc
```

**Pull request closed without merging**